src/stores/modal/IModalReducerState.ts

```diff
@@ -0,0 +1,6 @@
+interface IModelReducerState {
+    readonly currentModal: JSX.Element;
+    readonly modals: JSX.Element[];
```

Comment:
still not a fan of `modals` here, but you know how I feel about verbosity.

My head leans towards something like `modalList` or `modalCollection`.  This is purely a personal style thing, though, so take it for what it's worth.

Reply:
nice suggestion